### DISCUSSION TITLE
Is CRDT Happening? Offline?

### DISCUSSION THREAD

**Discussion by jtkeyva** [supabase/supabase#17496] [Feature Requests]
CRDT & Offline capability seem to be a highly requested feature and another nail in the coffin for Firebase.

You mentioned it on https://supabase.com/blog/postgres-crdt
and the repo https://github.com/supabase/pg_crdt which has not been updated in 9 months

Any plans to support this anytime soon?

In 20 years everything will be synced automatically...might as well get started now and this seems to be the way :)

**Comment by colinchadwick** [comment]
An update on that would be great. @kiwicopple

**Comment by cachapa** [comment]
I've been working with CRDTs for a while and wrote Dart wrappers around [Postgres](https://github.com/cachapa/postgres_crdt), [Sqlite](https://github.com/cachapa/sqlite_crdt), [Hive](https://github.com/cachapa/hive_crdt) (a Dart-native record store) and even a plain old [hashmap](https://github.com/cachapa/crdt).

Tying it all together is [crdt_sync](https://github.com/cachapa/crdt_sync) which does all the heavy lifting around connecting and synchronizing remote nodes.

My long-term goal is to make my work compatible with existing solutions, especially open-source ones like Supabase and ElectricSQL. I'm curious to know if you have a roadmap for this work, and if there's any interest in open collaboration.
